docs/rfcs/0003-rfc-controller_refactor.md

```diff
@@ -0,0 +1,111 @@
+
+One exception to the rule is `ConnectionHealth::ConnectionStability::Unstable`. The connection health object periodically sends pings and measures the received responses to assess whether or not the network connectivity is stable. To continue notifying users about their unstable network, we will continue to use this state within `ConnectionManager`. 
+
+Beyond the connectivity checks, the Controller also manages how much time has elapsed since the VPN was toggled on. Having this logic in the controller bloats the code without any gains, so that code will be moved out of `Controller` and instead a simple `Timer` object will encapsulates the related logic. The timer starts when the VPN enters State On since doing the initial checks may take a few seconds that we don't need to account for.
+
+Because this is all part of a unified connectivity experience, we do not want to release bits and pieces of this until everything is implemented, tested, and ready to go, this means that all implementation will live behind a feature flag until we are ready to test and release.
+
+
+## Implementation Plan
+Introduce a new `ConnectionManager` component that is activated when the client is launched and connects to `networkWatcher` in case anything changes. This new component will manage the following work:
+- Internet probe: assess if there is a network connection.
+- Firewall: assess if firewall is limiting network connectivity.
+- Captive portal check: determine uf the network is behind a captive portal.
+- Server probe: upon activation and then priodically check if there are available servers within user's selection location.
+- Subscription status: ensure that user has an active subscription, and if not, show the "No Subscription Found" screen.
+
+The logic to calculate elapsed time since the VPN was activated will also be moved out of the Controller and into a new `Timer` component. This bit of work will likely require a separate RFC. 
```

Comment:
```suggestion
The logic to calculate elapsed time since the VPN was activated will also be moved out of the Controller and into a new `Timer` component. This bit of work is out of scope for this RFC. 
```